src/main/java/net/mcreator/workspace/elements/VariableType.java

```diff
@@ -40,10 +45,28 @@ public String getName() {
 		return name;
 	}
 
-	public int getColor() {
+	public String getColor() {
 		return color;
 	}
 
+	/**
+	 * Returns the color of the blocks associated with this variable type. If the field is a valid hex color code, it's
+	 * returned as-is. If it's a valid integer, it's treated as a hue to get the color with the correct saturation and
+	 * value.
+	 * @return The color of the associated blocks, or black if it's badly formatted.
+	 */
+	public Color getBlocklyColor() {

```

Comment:
Wouldn't be better to move this code to `getColor` as it's its purpose? Adding another method to do the job is pretty useless.

Reply:
I still need the original color without changes to define the variable blocks, and also for consistency with the other value getters